sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -133,11 +137,45 @@ export class ApplicationStudentsController extends BaseController {
   })
   @ApiBadRequestResponse({ description: "Form validation failed." })
   @ApiNotFoundResponse({ description: "Application not found." })
+  @ApiForbiddenResponse({
+    description: "You have a restriction on your account.",
+  })
   async submitApplication(
     @Body() payload: SaveApplicationDto,
     @Param("applicationId") applicationId: number,
-    @UserToken() userToken: IUserToken,
+    @UserToken() studentToken: StudentUserToken,
   ): Promise<void> {
+    let isRestrictionActionExists = false;
```

Comment:
Can we move all this new logic to check the restriction to the application controller service?
This method is already too long and moving this can help with the code readability.
The method could be something like:
```typescript
this.applicationControllerService.checkRestrictions(studentId, payload.data.howWillYouBeAttendingTheProgram);
```

Reply:
i agree, this methods is long, i will move